setup.py

```diff
@@ -164,35 +168,50 @@ def wrapper(*args, **kw):  # type: ignore
     return wrapper
 
 
+_compute_shuffled_index = compute_shuffled_index
+compute_shuffled_index = cache_this(
+    lambda index, index_count, seed: (index, index_count, seed),
+    _compute_shuffled_index, lru_size=SLOTS_PER_EPOCH * 3)
+
+_get_total_active_balance = get_total_active_balance
+get_total_active_balance = cache_this(
+    lambda state: (state.validators.hash_tree_root(), state.slot),
+    _get_total_active_balance, lru_size=10)
+
 _get_base_reward = get_base_reward
 get_base_reward = cache_this(
-    lambda state, index: (state.validators.hash_tree_root(), state.slot),
-    _get_base_reward)
+    lambda state, index: (state.validators.hash_tree_root(), state.slot, index),
+    _get_base_reward, lru_size=10)
```

Comment:
Why size 10? Seems like it's valuable to have it be of at least the size of the validator set, right?

Reply:
Ah good catch. This is a remainder of other optimizations. When you traverse the pre-computed attester statuses once, you only need to re-use base reward very short-lived. But not in the current spec, this needs to have much more capacity indeed.